uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp

```diff
@@ -254,36 +248,39 @@ DataType CanopenInterface::sendQuery(RuntimeQuery query, uint8_t subindex) {
 
   ssize_t bytes_written = write(roboteq::CanopenInterface::socket_handle_, &query_frame, sizeof(struct can_frame));
   if (bytes_written != sizeof(struct can_frame)) {
-    // TODO: throw error
-    return 0;
+    throw std::runtime_error("Written packet size does not match can frame size");
   }
 
   struct can_frame response_frame = {};
   ssize_t bytes_read = read(roboteq::CanopenInterface::socket_handle_, &response_frame, sizeof(struct can_frame));
 
-  //  std::cout << std::hex << response_frame.can_id << "\t" << static_cast<unsigned>(response_frame.can_dlc) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[0]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[1])
-  //            << "\t" << static_cast<unsigned>(response_frame.data[2]) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[3]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[4])
-  //            << "\t" << static_cast<unsigned>(response_frame.data[5]) << "\t"
-  //            << static_cast<unsigned>(response_frame.data[6]) << "\t" <<
-  //            static_cast<unsigned>(response_frame.data[7])
-  //            << std::endl;
+   std::cout << std::hex << response_frame.can_id << "\t" 
+             << static_cast<unsigned>(response_frame.can_dlc) << "\t"
+             << static_cast<unsigned>(response_frame.data[0]) << "\t" <<
+             static_cast<unsigned>(response_frame.data[1])
+             << "\t" << static_cast<unsigned>(response_frame.data[2]) << "\t"
+             << static_cast<unsigned>(response_frame.data[3]) << "\t" <<
+             static_cast<unsigned>(response_frame.data[4])
+             << "\t" << static_cast<unsigned>(response_frame.data[5]) << "\t"
+             << static_cast<unsigned>(response_frame.data[6]) << "\t" <<
+             static_cast<unsigned>(response_frame.data[7])
+             << std::endl;
 
   if (bytes_read != sizeof(struct can_frame)) {
-    // TODO: throw error
-    return 0;
+    throw std::runtime_error("Read packet size does not match can frame size");
+  }
+
+  if (static_cast<unsigned>(response_frame.can_dlc) != CAN_FRAME_SIZE_BYTES) {
+    throw std::runtime_error("Mismatched DLC value");
   }
 
   // TODO: throw appropriate errors
   if ((response_frame.data[0] & RESPONSE_TYPE_MASK) != SUCCESSFUL_QUERY_RESPONSE) {
-    std::cout << "Query unsuccessful response" << std::endl;
+    throw std::runtime_error("Unssuccessful query response");
   } else if (query_frame.data[1] != response_frame.data[1] || query_frame.data[2] != response_frame.data[2]) {
```

Comment:
i believe one of your clang tidy failures is calling else after throws. This makes code harder to follow. 

It should be smth list:
```
if (error){
    throw error;
}

if (ohter_error){
    throw other_error;
}
```

and no need for else case cuz its like just the default if the block hasnt thrown an error